backend/src/auth.rs

```diff
@@ -76,28 +76,176 @@ impl RequiredUser {
     /// `require-verification` (true/false)
     ///
     /// If `require-verification` is set to true, then the user account must be verified
-    pub(crate) fn is_special(&self, config: &crate::Config) -> bool {
-        if config.allowed_emails.is_empty() {
-            ft_sdk::println!(
-                "`allowed-emails` list is empty. No one is allowed to create meetings"
-            );
-            return false;
+    pub(crate) fn is_special(
+        &self,
+        config: &crate::Config,
+        host: &ft_sdk::Host,
+    ) -> Result<(), AuthorizationError> {
+        let allowed_emails = get_allowed_emails_or_host(config.allowed_emails.clone(), host)?;
+
+        let email_matches = allowed_emails
+            .split(',')
+            .map(|v| v.trim())
+            .any(|v| !v.is_empty() && self.email.ends_with(v));
+
+        if !email_matches {
+            return Err(AuthorizationError::Unauthorized);
         }
 
-        config.allowed_emails.split(',').map(|v| v.trim()).any(|v| {
-            if v.is_empty() {
-                return false;
-            }
+        if config.require_verification && !self.email_is_verified {
+            return Err(AuthorizationError::RequiresVerification);
+        }
 
-            // check email domain
-            let email_matches = self.email.ends_with(v);
-            // if the email matches, check if the account is verified (verification is done
-            // by clicking on a link that is sent to user's email)
-            if config.require_verification {
-                return email_matches && self.email_is_verified;
-            }
+        Ok(())
+    }
+}
 
-            email_matches
-        })
+/// Get the allowed emails from the `who-can-create-meetings` ftd variable or the request host if
+/// it's empty
+/// Returns [AuthorizationError::EmptyWhitelist] if both are empty
+fn get_allowed_emails_or_host(
+    allowed_emails: String,
+    host: &ft_sdk::Host,
+) -> Result<String, AuthorizationError> {
+    if !allowed_emails.is_empty() {
+        return Ok(allowed_emails);
+    }
+
+    let domain = match host
+        .without_port()
+        .split('.')
+        .collect::<Vec<&str>>()
+        .as_slice()
+    {
+        // meet.fifthtry.com
+        [_, host, ext] => Some(format!("{}.{}", host, ext)),
```

Comment:
This will not work with `foo.bar.gov.in` for example. Use `[.., host, ext]` instead.

Reply:
https://github.com/fifthtry-community/lets-talk/pull/7/commits/17b5c7aea1e72ca3d4f9e2b18275978cb7033eba This is probably what you meant?